app/scripts/lib/setupAccountLabelsPetnamesBridge.ts

```diff
@@ -0,0 +1,105 @@
+import { NameController, NameType } from '@metamask/name-controller';
+import { cloneDeep } from 'lodash';
+import log from 'loglevel';
+
+export type IdentityEntry = {
+  address: string;
+  name: string;
+};
+
+export type PertinentState = {
+  identities: { [address: string]: IdentityEntry };
+};
+
+export type PertinentPreferencesController = {
+  store: {
+    getState: () => PertinentState;
+    subscribe: (callback: (state: PertinentState) => void) => void;
+  };
+};
+
+const MAINNET_CHAIN_ID = '0x1';
+export const ACCOUNT_LABEL_NAME_TYPE = NameType.ETHEREUM_ADDRESS;
+export const ACCOUNT_LABEL_CHAIN_ID = MAINNET_CHAIN_ID;
+
+/**
+ * Groups the entries in the old and new entries arrays into added, updated and
+ * deleted entries, like a patch.
+ *
+ * @param oldEntries - The last seen IdentityEntries in the preferences controller.
+ * @param newEntries - The new IdentityEntries.
+ */
+function groupEntries(
```

Comment:
As this logic is very similar to the `AddressBookPetnamesBridge`, is it worth creating an abstract `PetnamesBridge` which we could re-use for both the address book and account identities to simplify them both and reduce duplication, plus simplify the unit testing.

It could just be the existing `AddressBookPetnamesBridge` logic but with the source of the data abstracted through abstract methods such as `onDataSourceChange`, `getSourceEntries`, and `updateSourceEntries`.

We could also use a constructor argument such as `isBidirectional` to control whether a specific bridge cares about updating the source, as done with the address book but not the account identities.